test/power_shell_SUITE.erl

```diff
@@ -485,6 +486,21 @@ export_no_link(Config) when is_list(Config) ->
             ?assertException(error, undef, power_shell_export:local_unexported(success))
     end.
 
+recursive_eval() ->
+    [{doc, "Tests on_load preserves stack"}].
+
+recursive_eval(Config) ->
+    Source =
+        "-module(recursive). -export([foo/0, bar/0]). "
+        "foo() -> ok. bar() -> power_shell:eval(recursive, foo, []), true.",
+    PrivPath = ?config(priv_dir, Config),
+    true = code:add_path(PrivPath),
+    Filename = filename:join(PrivPath, "recursive.erl"),
+    ok = file:write_file(Filename, Source),
+    ?assert(power_shell:eval(recursive, bar, [])),
+    true = code:del_path(PrivPath),
```

Comment:
I'd also add `?assertEqual(PrevDict, get())` (given that PrevDict = get() before the `eval` call).